src/overlays/actors/ovl_En_Egblock/z_en_egblock.c

```diff
@@ -28,28 +41,185 @@ const ActorInit En_Egblock_InitVars = {
     (ActorFunc)EnEgblock_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_06001698;
-extern UNK_TYPE D_06001820;
-extern UNK_TYPE D_06001BF8;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/EnEgblock_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/EnEgblock_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/func_80ABA8A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/func_80ABA988.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/func_80ABA9B8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/EnEgblock_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/EnEgblock_Draw.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/func_80ABAAF4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/func_80ABACB4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egblock/func_80ABAE64.s")
+void EnEgblock_Init(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    CollisionHeader* colHeader = NULL;
+    EnEgblock* this = THIS;
+
+    DynaPolyActor_Init(&this->dyna, 0);
+    CollisionHeader_GetVirtual(&gEyegoreBlockColHeader, &colHeader);
+    this->dyna.bgId = DynaPoly_SetBgActor(play, &play->colCtx.dyna, &this->dyna.actor, colHeader);
+    this->dyna.actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->inactive = false;
+    Actor_SetScale(&this->dyna.actor, 0.5f);
+    this->type = GET_EGBLOCK_TYPE(&this->dyna.actor);
+    this->param1 = GET_EGBLOCK_PARAM_F80(&this->dyna.actor);
+    this->param2 = GET_EGBLOCK_PARAM_7F(&this->dyna.actor);
+    if (this->type == EGBLOCK_TYPE_ACTIVE) {
+        this->dyna.actor.colChkInfo.health = 1;
+        this->actionFunc = EnEgblock_Active;
+    } else {
+        this->actionFunc = EnEgblock_Static;
+    }
+}
+
+void EnEgblock_Destroy(Actor* thisx, PlayState* play2) {
+    EnEgblock* this = THIS;
+    PlayState* play = play2;
+
+    if (this->dyna.actor.colChkInfo.health == 1) {
+        DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+void EnEgblock_Active(EnEgblock* this, PlayState* play) {
+    if (this->dyna.actor.colChkInfo.health <= 0) {
+        s32 i;
+
+        for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+            EnEgblock_SpawnEffect(this, &this->dyna.actor.world.pos, 30, (i & 1) + EGBLOCK_EFFECT_DEBRIS_FLAT);
+            //! @bug Egblock doesn't have an effect with index 2, so half of these spawns do nothing.
```

Comment:
IMO, this should go *above* the previous line rather than below it like you have, but it's a matter of opinion.

Reply:
Agreed, we usually put bug comments above the code that causes the bug.

Reply:
Addressed?